nifi-nar-bundles/nifi-zendesk-bundle/nifi-zendesk-processors/src/main/java/org/apache/nifi/processors/zendesk/PutZendeskTicket.java

```diff
@@ -0,0 +1,294 @@
+            final String subject = context.getProperty(ZENDESK_TICKET_SUBJECT).evaluateAttributeExpressions().getValue();
+            final String priority = context.getProperty(ZENDESK_TICKET_PRIORITY).evaluateAttributeExpressions().getValue();
+            final String type = context.getProperty(ZENDESK_TICKET_TYPE).evaluateAttributeExpressions().getValue();
+            final Map<String, String> dynamicProperties = getDynamicProperties(context, context.getProperties());
+            List<ObjectNode> zendeskTickets = new ArrayList<>();
+
+            try (final InputStream in = session.read(flowFile); final RecordReader reader = readerFactory.createRecordReader(flowFile, in, getLogger())) {
+                Record record;
+
+                while ((record = reader.nextRecord()) != null) {
+                    ObjectNode baseTicketNode = mapper.createObjectNode();
+                    resolveFieldValue("/comment/body", commentBody, baseTicketNode, record);
+
+                    if (subject != null) {
+                        resolveFieldValue("/subject", subject, baseTicketNode, record);
+                    }
+                    if (priority != null) {
+                        resolveFieldValue("/priority", priority, baseTicketNode, record);
+                    }
+                    if (type != null) {
+                        resolveFieldValue("/type", type, baseTicketNode, record);
+                    }
+                    for (Map.Entry<String, String> dynamicProperty : dynamicProperties.entrySet()) {
+                        addDynamicField(dynamicProperty.getKey(), dynamicProperty.getValue(), baseTicketNode, record);
+                    }
+                    zendeskTickets.add(baseTicketNode);
+                }
+
+            } catch (IOException | SchemaNotFoundException | MalformedRecordException e) {
+                getLogger().error("Error occurred while creating zendesk tickets", e);
+                session.transfer(session.penalize(flowFile), REL_FAILURE);
+                return;
+            }
+
+            if (zendeskTickets.size() == 0) {
+                getLogger().info("No record were received");
+                return;
+            }
+
+            try {
+                final InputStream inputStream = createRequestObject(mapper, zendeskTickets);
+                uri = createUri(authenticationContext, zendeskTickets.size());
+                response = performPostRequest(authenticationContext, webClientServiceProvider, uri, inputStream);
+            } catch (IOException e) {
+                getLogger().error("Failed to post request to Zendesk", e);
+                session.transfer(session.penalize(flowFile), REL_FAILURE);
+                return;
+            }
+        }
+
+        handleResponse(session, flowFile, response, uri, startNanos);
+    }
+
+    private void handleResponse(ProcessSession session, FlowFile flowFile, HttpResponseEntity response, URI uri, long startNanos) {
```

Comment:
Please add the corresponding mimetype to the FlowFile which should be the mime type of the record reader or JSON.